Source/WebCore/css/StyleProperties.cpp

```diff
@@ -705,6 +707,68 @@ String StyleProperties::fontVariantValue() const
     return result.toString();
 }
 
+String StyleProperties::fontSynthesisValue() const
+{
+    StringBuilder result;
+
+    auto getExplicitLonghandValue = [&](CSSPropertyID propertyID) -> CSSValue* {
+        auto foundPropertyIndex = findPropertyIndex(propertyID);
+        if (foundPropertyIndex == -1)
+            return nullptr;
+
+        auto property = propertyAt(foundPropertyIndex);
+        if (property.isImplicit())
+            return nullptr;
+
+        return property.value();
+    };
+
+    auto appendWithPrefixIfNeeded = [&](ASCIILiteral word) {
+        if (!result.isEmpty())
+            result.append(' ');
+        result.append(word);
+    };
+    
+    // font-synthesis: none | [ weight || style || small-caps ]
+    auto weightValue = getExplicitLonghandValue(CSSPropertyFontSynthesisWeight);
+    auto styleValue = getExplicitLonghandValue(CSSPropertyFontSynthesisStyle);
+    auto capsValue = getExplicitLonghandValue(CSSPropertyFontSynthesisSmallCaps);
+
+    auto valueID = [&](CSSValue *value) {
+        if (!value)
+            return CSSValueInvalid;
```

Comment:
```suggestion
        if (!value || !is<CSSPrimitiveValue>(value))
            return CSSValueInvalid;
```
to avoid crashing if we don't have a `CSSPrimitiveValue`.